x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -68,139 +41,191 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
-		Sender:          sender.Address.String(),
-		LiquidityAmount: withdrawAmountInt,
+		PositionId:      position.PositionId,
+		Sender:          position.GetAddress(),
+		LiquidityAmount: withdrawAmount,
 	}, nil
 }
 
 func RandMsgCollectFees(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCollectFees, error) {
-	rand := sim.GetRand()
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	poolI, err := k.GetPool(ctx, clPool.GetId())
+	if err != nil {
+		return nil, err
+	}
+
+	// positionCreator creates the position with pool denoms
 	// get random user address with the pool denoms
 	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
 	if !senderExists {
 		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
+	// get random user positions
+	positions, err := k.GetUserPositions(ctx, sender.Address, clPool.GetId())
 	if err != nil {
 		return nil, fmt.Errorf("position does not exist")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	var positionIds []uint64
+	for _, position := range positions {
+		positionIds = append(positionIds, position.PositionId)
+	}
+
+	// SwapOwner performs the swap from token0 to token1
+	swapOwner, swapOwnerTokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we always have 2 tokens
+	// Note: tokens returns a random subset of poolDenoms, so  had to add this assertion
+	if len(swapOwnerTokens) < 2 {
+		return nil, fmt.Errorf("user does not have pool tokens")
 	}
 
-	// pick a random position
-	randPosition := positions[rand.Intn(len(positions))]
+	// perform swap until token 1 runs out
+	remainingSwapOwnerToken0Amt := swapOwnerTokens[0].Amount
+	remainingSwapOwnerToken1Amt := swapOwnerTokens[1].Amount
+	for remainingSwapOwnerToken0Amt.ToDec().TruncateInt().GT(sdk.ZeroInt()) && remainingSwapOwnerToken1Amt.ToDec().TruncateInt().GT(sdk.ZeroInt()) {
+		randToken0Amt := sim.RandomAmount(remainingSwapOwnerToken0Amt)
+		randToken1Amt := sim.RandomAmount(remainingSwapOwnerToken1Amt)
+
+		if randToken0Amt.LTE(sdk.ZeroInt()) || randToken1Amt.LTE(sdk.ZeroInt()) {
+			return nil, fmt.Errorf("invalid amount to swap")
+		}
+
+		// perform swap from token0 to token1 until either token0 or token1 fund runs out
+		_, err = k.SwapExactAmountIn(ctx, swapOwner.Address, poolI, sdk.NewCoin(swapOwnerTokens[0].Denom, randToken0Amt), swapOwnerTokens[1].Denom, sdk.OneInt(), sdk.NewDecWithPrec(1, 2))
+		if err != nil {
+			return nil, err
+		}
+
+		remainingSwapOwnerToken0Amt = remainingSwapOwnerToken0Amt.Sub(randToken0Amt)
+		remainingSwapOwnerToken1Amt = remainingSwapOwnerToken1Amt.Sub(randToken1Amt)
+	}
```

Comment:
I am not quite understanding the logic here. Why are we swapping until one of the tokens runs out?

Reply:
this is basically swapping until the owner runs out of funds, i removed token1Amt calculations to make things clearer. So now, user swaps tokenA -> tokenB until their tokenA funds are <= 0